### PR TITLE
Ret 2155

### PR DESCRIPTION
**Pull request by btensay**
An implementation of "Delete Judge" event  covering controller endpoint and JudgeService extension. Unit tests added too.

https://tools.hmcts.net/jira/browse/RET-2155

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/admin/staff/JudgeController.java [hunk 1]
```diff
@@ -152,4 +152,72 @@ public ResponseEntity<CCDCallbackResponse> updateJudge(
...
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "Accessed successfully"),
+        @ApiResponse(responseCode = "400", description = "Bad Request"),
+        @ApiResponse(responseCode = "500", description = "Internal Server Error")
+    })
+    public ResponseEntity<CCDCallbackResponse> deleteJudgeMidEventSelectOffice(
+        @RequestHeader("Authorization") String userToken,
+        @RequestBody CCDRequest ccdRequest) {
+
+        log.info("/deleteJudgeMidEventSelectOffice");
```
Comment (gauravhmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/admin/staff/JudgeController.java:166]: May be we don't need this log message here?

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/admin/staff/JudgeService.java [hunk 2]
```diff
@@ -119,4 +114,60 @@ public List<String> updateJudge(AdminData adminData) {
 
         return errors;
     }
+
+    public List<String> deleteJudge(AdminData adminData) {
+        List<String> errors = new ArrayList<>();
+        var selectedId = Integer.parseInt(adminData.getJudgeSelectList().getSelectedCode());
```
Comment (gauravhmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/admin/staff/JudgeService.java:120]: Avoid using vars. We don't like them anymore.

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/admin/staff/JudgeController.java [hunk 3]
```diff
@@ -152,4 +152,72 @@ public ResponseEntity<CCDCallbackResponse> updateJudge(
...
+
+        if (!verifyTokenService.verifyTokenSignature(userToken)) {
+            return ResponseEntity.status(HttpStatus.FORBIDDEN.value()).build();
+        }
+
+        var adminData = ccdRequest.getCaseDetails().getAdminData();
```
Comment (hsjhita) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/admin/staff/JudgeController.java:172]: Same as Gaurav - avoid using var now.

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/admin/staff/JudgeService.java [hunk 4]
```diff
@@ -119,4 +114,60 @@ public List<String> updateJudge(AdminData adminData) {
 
         return errors;
     }
+
+    public List<String> deleteJudge(AdminData adminData) {
```
Comment (hsjhita) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/admin/staff/JudgeService.java:118]: Java Docs

Comment (hsjhita) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/admin/staff/JudgeService.java:118]: same for other public methods

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/admin/staff/JudgeController.java (hunk 3 above)
Comment (nickwest1) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/admin/staff/JudgeController.java:172]: Same applies throughout this PR ;)

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/admin/staff/JudgeServiceTest.java [hunk 5]
```diff
@@ -178,18 +260,21 @@ private AdminData createAdminData(String judgeCode, String judgeName, String tri
...
         var adminData = createAdminData(code, name, tribunalOffice, employmentStatus);
-        adminData.setTribunalOffice(tribunalOffice);
+        adminData.setJudgeSelectList(getDynamicFixedListType(id, name));
+        adminData.getJudgeSelectList().setValue(getDynamicValueType(id, name));
+        return adminData;
+    }
+
+    private DynamicValueType getDynamicValueType(String id, String name) {
+        return DynamicValueType.create(id, name);
+    }
```
Comment (nickwest1) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/admin/staff/JudgeServiceTest.java:270]: Having this in a separate method adds no value

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/admin/staff/JudgeService.java [hunk 6]
```diff
@@ -57,46 +66,52 @@ public void saveJudge(AdminData adminData) {
...
         return errors;
     }
 
+    /**
+     * Updates the details of the selected Judge.
+     * @param adminData The object containing admin data
+     */
     public List<String> updateJudge(AdminData adminData) {
         List<String> errors = new ArrayList<>();
         var selectedId = Integer.parseInt(adminData.getJudgeSelectList().getSelectedCode());
```
Comment (gauravhmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/admin/staff/JudgeService.java:117]: And I finally found 1 var!

**Review by gauravhmcts — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/admin/staff/JudgeService.java (hunk 6 above)
Comment (btensay) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/admin/staff/JudgeService.java:117]: Sorted.

**Review by hsjhita — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/admin/staff/JudgeServiceTest.java [hunk 7]
```diff
@@ -177,23 +262,22 @@ private AdminData createAdminData(String judgeCode, String judgeName, String tri
...
                                                      String employmentStatus) {
-        var adminData = createAdminData(code, name, tribunalOffice, employmentStatus);
-        adminData.setTribunalOffice(tribunalOffice);
+        AdminData adminData = createAdminData(code, name, tribunalOffice, employmentStatus);
+        adminData.setJudgeSelectList(getDynamicFixedListType(id, name));
+        adminData.getJudgeSelectList().setValue(DynamicValueType.create(id, name));
+        return adminData;
+    }
 
+    private DynamicFixedListType getDynamicFixedListType(String id, String name) {
```
Comment (jthmcts) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/admin/staff/JudgeServiceTest.java:271]: You can probably replace this method with a call to the DynamicFixedListType static factory method .of

Comment (btensay) [reply src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/admin/staff/JudgeServiceTest.java]: Done

**Pull request merged**